### ISSUE TITLE
material in tracks/ not GPL compliant

### ISSUE THREAD

**Issue by bkuhn** [usrbinnc/netcat-cpi-kernel-module#8]
Since Linux modules must be released under the GPL, all sources must be GPL-compliant.  I don't believe that the material in tracks/ is in ["preferred form of the work for making modifications to it"](http://www.gnu.org/licenses/gpl-2.0.html#section3).  This could be remedied by providing the audio files that were used to generate the trck*data.h files, and the commands used to generate them.

Thanks for your time and effort to remedy this GPL violation!

**Comment by jmtd** [comment]
> Since Linux modules must be released under the GPL,

I think you're wrong, there. Certainly no modules will be _merged_ unless they're GPL-2, but 3<sup>rd</sup> parties are fine to write and distribute non-GPL modules.

**Comment by dbalatero** [comment]
> I think you're wrong, there. Certainly no modules will be merged unless they're GPL-2, but 3rd parties are fine to write and distribute non-GPL modules.

Could the source file be GPL, but the track files be Creative Commons somehow? We're not really interested in truly merging this into mainline or anything, for obvious reasons :)

**Comment by blucia0a** [comment]
Yea, I think our licensing situation is probably like the one for binary device drivers -- the tracks are analogous to NVidia's proprietary binary driver parts.  That said, they arent quite proprietary like a driver because there isnt really a source to distribute, that we're opting not to (except the 24 track tapes, which our studio has).

**Comment by stillinbeta** [comment]
It seems to me like you could just specificy a non-free license using [MODULE_LICENSE](http://tldp.org/LDP/lkmpg/2.6/html/lkmpg.html#AEN279), no? Then you wouldn't need to worry about the hybrid license or whatever.

**Comment by dbalatero** [comment]
Something like `MODULE_LICENSE("Proprietary")`?

**Comment by bkuhn** [comment]
I'm not sure you want to emulate the behavior of GPL violators like Nvidia,

Since many Linux copyright holders do indeed believe that Linux is GPL'd, the license of Linux is GPL'd.  Publication of modules designed specifically for Linux is a combined work, and is covered by GPL.  I assume that your goal here is to be friendly to the whole Linux community.

Could you perhaps dual license the music under (Some-CC-License|GPLv2-only)?  That would solve the issue.

MODULE_LICENSE, BTW, is merely a statement that doesn't change the combined works issues.  I would even argue that MODULE_LICENSE("Proprietary") is the code equivalent of shouting from the rooftops: "Hey, look over here, I'm violating the GPL.  Check me out." :)

**Comment by wxl** [comment]
> Could you perhaps dual license the music under (Some-CC-License|GPLv2-only)?

+1!!!

**Comment by blucia0a** [comment]
Thanks for helping us figure this out!  I think we're converging to something that is GPLv2-only + CC for the track data.  We're going to meet as a band tonight and get something installed ASAP.

**Comment by wxl** [comment]
> I think we're converging to something
> that is GPLv2-only + CC for the track data.

Don't forget to CC the Bandcamp!

**Comment by gregkh** [comment]
This is "firmware" blobs, there is no intrepreting of the data in the kernel, so the music should not need to be released under any GPL-like license.  We have lots of examples of this in the kernel source tree.  Just allow it to be redistributed freely, in the format it currently is in, and all should be fine.

**Comment by bkuhn** [comment]
@gregkh, I don't think there's an explicit GPL exception for firmware blobs in any event, so clarifying the situation is probably the best step.

Also, there are substantive differences between this and the firmware blob situation, particularly since firmware blobs expressed purpose is to encode a separate program in a .h file with the only purpose being to install it on hardware with its own processor that Linux doesn't actually run on at all.

Anyway, I think the goal here of the band is to do the right thing by leaps and bounds, not by "barely eeking it out", and the right solution for that is (CC|GPLv2-only) the music and provide the music "source' files.

**Comment by gregkh** [comment]
@bkuhn you and I differ on the firmware issue, and that's fine.

And I see this exactly like firmware, the kernel is doing no processing on the data at all except streaming it through to userspace, where other programs, with other licenses, are doing things with the data.

But as neither of us are lawyers, just people who deal with this all the time, odds are we are just confusing the authors of the code/music, which isn't a good thing to do here :)

**Comment by bkuhn** [comment]
@gregkh , agreed.  This band is having fun doing something entertaining with Linux, so we should absolutely encourage it.  And, our side issue here is moot for them anyway since they have decided to take my suggestion. :)  We don't need to have yet another complex combined works debate on _this_ thread too.  We have it in enough fora as it is. :)

**Comment by blucia0a** [comment]
I sure am confused!  We're going to discuss it tonight as a band and we'll probably come up with a solution in the next day or so.  Thanks a lot for your input and contributions to the project.  We really appreciate the interest in our album release!

**Comment by bkuhn** [comment]
@blucia0a, short version is: I argue something needs to be done with the licensing of material in tracks/, @gregkh argues you don't.  You will be beyond reproach if you make a link to the "source" files of the music and dual license it with GPL (e.g., (CC-By-SA|GPL) ).  @gregk would argue this isn't necessary, I would argue it is. :)  Does that clear up confusion?

**Comment by blucia0a** [comment]
@bkuhn Yes, thanks for the summary - that helped, too.  I don't have a strong opinion on this issue, like it seems you and @gregkh both do.  I'm hoping we can settle with something that is a satisfying middle ground, appropriate for this kind of software/music release.

**Comment by ClashTheBunny** [comment]
IANAL, but there are many modules that 'taint' the kernel with a non-GPL licence.  As long as they distribute it in source form that are not breaking the GPL.  I thought that most everybody makes a distinction about linking vs loading.  The songs are actually combined into a binary with the song data because they are in .h files and not loaded from /(usr/)/lib/firmware.  Licencing the code under GPL would make the music GPL also.

The thing that the band should do is license it.  Distributing things without a license makes many nerds nervous.  Something like the CC-NC-BY-ND would give people the right to get the word out without the band having to fear a bunch of people making and selling knockoffs.

The code can then be something with a soft copyleft licence such as apache, BSD or MIT. The hard copy left GPL would possibly be a mistake as far as linking goes.

**Comment by wxl** [comment]
> The thing that the band should do is license it. Distributing things without a
> license makes many nerds nervous. Something like the CC-NC-BY-ND would give
> people the right to get the word out without the band having to fear a bunch of
> people making and selling knockoffs.

As a netlabel owner and also the station manager at a small internet
radio station, I would advise against any of the no derivative (ND)
licenses from Creative Commons. The general consensus among the CC
community is that one should assume the conservative interpretation when
no other information is provided, just like copyright is assumed when
there is no license. That being said, no derivatives means just that,
including DJ mixes. So unless people play the song in entirety and end
it completely before starting another, it limits airplay which can have
an effect on marketing.

I instead suggest CC BY-NC-SA. It prohibits people from selling their
knock offs (read: remixes) and also ensures that when they share or
create derivatives, they would need to use a similar license or Share
Alike (SA).

If you want to get all Stallman on it, you'd go with CC BY-SA.

Honestly, we're all for reusing code, why would we not do it with music?
Isn't it all just data?

/me steps off the soapbox :)

**Comment by jmtd** [comment]
@wxl, CC NC has similar problems. If you play it on a commercial radio station, some would argue that's commercial use. If someone bundles it up with a bunch of other software and distributes that on DVDs and sells the DVDs to cover the costs of distribution, some would argue _that's_ commercial use. etc.

@bkuhn:

> Since many Linux copyright holders do indeed believe that Linux is GPL'd, the license of Linux is GPL'd.

This is a confusing statement. There's no debate that the kernel is GPL'd.

> Publication of modules designed specifically for Linux is a combined work

I realise you've clarified in later comments your position versus others, including @gregkh's, but I found this sentence to be misleading. It's presented as fact when there's no consensus on this, and runs on from the first statement which is obvious, as if to suggest a logical implication.

**Comment by wxl** [comment]
> @wxl, CC NC has similar problems. If you play it on a commercial radio station,
> some would argue that's commercial use. If someone bundles it up with a bunch
> of other software and distributes that on DVDs and sells the DVDs to cover the
> costs of distribution, some would argue that's commercial use. etc.

I agree wholeheartedly. I prefer CC BY-SA frankly. I don't worry so much
about people making money off my stuff. It's free marketing. ;)

**Comment by bkuhn** [comment]
@jmtd, I see your point about my statement being confusing.  I should say it this way: "Many Linux copyright holders have views about the GPL that don't match what the text of the license says and often just look the other way on GPL violations.  Other Linux copyright holders disagree and enforce GPL with regard to things like modules, firmwares, etc."  I hope that clarifies my point. Also, you mention "consensus" but note that consensus is moot here: enough Linux copyright holders with substantial code in Linux agree with my interpretations of GPLv2, so -- absent the other side writing their code entirely out of Linux and rejecting patches from people who have that interpretation -- the most conservative interpretation of GPLv2 is the one that matters.  But again, all this is part is off-topic to this ticket, IMO.

BTW, @ClashTheBunny, I think you are conflating GPL-compatible licenses with GPL-incompatible licenses in your statements.

@wxl et al, I agree that Cc-By-SA is the right choice for the Creative Commons side, but note that issue, while important, is orthogonal here.  If the band were to use a disjunctive license with GPLv2-only, that would solve the problem I was raising (provided CCS is provided for tracks/).  I hope they just license the whole thing (CC-By-SA-4.0-International|GPLv2-only) and include details of how to turn the audio into track/*.h files.  That would be the perfect outcome from my point of view.

**Comment by sarciszewski** [comment]
To the entire GPL argument: This is why we can't have nice things.

**Comment by jmtd** [comment]
@sarciszewski: if only there were fewer GPL arguments; we'd have more Eps-As-Kernel-Modules?

**Comment by bkuhn** [comment]
Scott Arciszewski wrote at 08:50 (EDT) on Thursday:

> To the entire GPL argument: This is why we can't have nice things.

I don't think that's what it's about here.  I think the band has been
open-minded to making the thing perfectly GPL-compliant, which makes it
even nicer.  That was always my goal here: let's not just have nice
things, but really nice things. :)

   -- bkuhn
